### PR TITLE
Force create dir on generate

### PR DESCRIPTION
**Pull request by Kareylo**
fix issue #52

### CODE REVIEW

File: tests/TestCase/Shell/PhpstormShellTest.php [hunk 1]
```diff
@@ -67,6 +67,26 @@ public function tearDown() {
 	/**
 	 * @return void
 	 */
+	public function testDirExists()
+	{
+		$this->assertSame(false, file_exists(TMP . '.meta.php'));
```
Comment (dereuromark) [inline tests/TestCase/Shell/PhpstormShellTest.php]: assertFalse and assertTrue

File: tests/TestCase/Shell/PhpstormShellTest.php [hunk 2]
```diff
@@ -67,6 +67,26 @@ public function tearDown() {
 	/**
 	 * @return void
 	 */
+	public function testDirExists()
```
Comment (dereuromark) [inline tests/TestCase/Shell/PhpstormShellTest.php]: You are actually testing the FileExists not DirExists here

Comment (Kareylo) [reply tests/TestCase/Shell/PhpstormShellTest.php]: file_exists check for both file and dir : http://php.net/manual/en/function.file-exists.php

Comment (Kareylo) [reply tests/TestCase/Shell/PhpstormShellTest.php]: I think we'll need to add a dir in TMP to put the file in

File: src/Shell/PhpstormShell.php [hunk 3]
```diff
@@ -91,12 +92,16 @@ protected function getMetaFilePath() {
...
-
 		return ROOT . DS . '.phpstorm.meta.php' . DS . '.ide-helper.meta.php';
 	}
 
+	/**
+	 * @return void
+	 */
+	protected function ensureDir() {
+		if (!file_exists(dirname($this->getMetaFilePath()))) {
+			mkdir(dirname($this->getMetaFilePath()), '0775', true);
```
Comment (dereuromark) [inline src/Shell/PhpstormShell.php]: 0755 is octal. dont use string here.

File: src/Shell/PhpstormShell.php [hunk 4]
```diff
@@ -94,4 +95,13 @@ protected function getMetaFilePath() {
 		return ROOT . DS . '.phpstorm.meta.php' . DS . '.ide-helper.meta.php';
 	}
 
+	/**
+	 * @return void
+	 */
+	protected function ensureDir() {
+		if (!file_exists(dirname($this->getMetaFilePath()))) {
+			mkdir(dirname($this->getMetaFilePath()), 0777, true);
```
Comment (dereuromark) [inline src/Shell/PhpstormShell.php]: does 775 suffice?

**Pull request merged**